modules/apps/frontend-js/frontend-js-walkthrough-web/test/js/Walkthrough.test.js

```diff
@@ -0,0 +1,251 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import '@testing-library/jest-dom/extend-expect';
+import {cleanup, render, screen} from '@testing-library/react';
+import userEvents from '@testing-library/user-event';
+import {navigate} from 'frontend-js-web';
+import React from 'react';
+
+import Walkthrough from '../../src/main/resources/META-INF/resources/Walkthrough';
+import {
+	BOX_SHADOW_ELEMENT_MOCK,
+	DOM_STRUCTURE_FOR_PLACING_STEPS,
+	INVALID_NODE_SELECTOR_MOCK,
+	MULTI_PAGES_MOCK,
+	PAGE_MOCK,
+} from './__fixtures__/walkthroughMock';
+
+function setupDocument() {
+	const domStructureReference = document.createElement('div');
+
+	domStructureReference.innerHTML = DOM_STRUCTURE_FOR_PLACING_STEPS;
+
+	domStructureReference.id = 'test-structure';
+
+	document.body.appendChild(domStructureReference);
+
+	return () => {
+		window.localStorage.clear();
+
+		const nodeToBeRemoved = document.getElementById('test-structure');
+
+		document.body.removeChild(nodeToBeRemoved);
+
+		nodeToBeRemoved.remove();
+	};
+}
+
+function renderWalkthrough(props) {
+	return render(<Walkthrough {...props} />, {
+		baseElement: document.getElementById('app_root'),
+	});
+}
+
+jest.mock('frontend-js-web', () => ({
+	navigate: jest.fn((url) => url),
+}));
+
+const USER_ID = 42;
+
+describe('Walkthrough', () => {
+	window.themeDisplay = {
+		getLayoutRelativeURL: jest.fn(() => '/home'),
+		getUserId: jest.fn(() => USER_ID),
+	};
+
+	afterEach(() => {
+		cleanup();
+	});
+
+	beforeEach(() => {
+		jest.restoreAllMocks();
+	});
+
+	it('renders', () => {
+		const cleanUp = setupDocument();
+
+		const {container} = renderWalkthrough(PAGE_MOCK);
+
+		expect(container).toBeInTheDocument();
+
+		cleanUp();
+	});
+
+	it(`when clicking on Next, it navigates to the given URL of 'next'`, () => {
+		const cleanUp = setupDocument();
+
+		const {getByLabelText} = renderWalkthrough(MULTI_PAGES_MOCK);
+
+		const hotspot = getByLabelText('start-the-walkthrough');
+
+		userEvents.click(hotspot);
+
+		userEvents.click(screen.getByText('ok'));
+
+		expect(navigate).toBeCalledWith(MULTI_PAGES_MOCK.steps[0].next);
+
+		cleanUp();
+	});
+
+	xit(`when clicking on Previous, it navigates to the given url of 'previous'`, () => {
+		const cleanUp = setupDocument();
+
+		const {getByLabelText} = renderWalkthrough(MULTI_PAGES_MOCK);
+
+		const hotspot = getByLabelText('start-the-walkthrough');
+
+		userEvents.click(hotspot);
+
+		userEvents.click(screen.getByText('ok'));
+
+		// Change the location to '/home-2' and then make the Walkthrough render on this new page :/
+		// I consider this inviable. Maybe we should force the step 2 to be on the current page but IDK
+		// yet how to do it
+
+		// userEvents.click(screen.getByText('previous'));
+		// expect(navigate).toBeCalledWith(MULTI_PAGES_MOCK.steps[1].previous);
```

Comment:
This was a leftover that I would work on tomorrow.

Reply:
https://github.com/liferay-frontend/liferay-portal/pull/2676/commits/2d21195189bb21dc548422ca68915436d66c6be4